lfs/batcher.go

```diff
@@ -39,11 +42,20 @@ func (b *Batcher) Next() []Transferable {
 }
 
 // Exit stops all batching and allows Next() to return. Calling Add() after
-// calling Exit() will result in a panic.
+// calling Exit() will result in a panic, unless Reset() is called first.
 func (b *Batcher) Exit() {
+	atomic.StoreUint32(&b.exited, 1)
 	close(b.input)
 }
 
+// Reset allows an Exit()ed batcher to be re-started.
+func (b *Batcher) Reset() {
+	if atomic.CompareAndSwapUint32(&b.exited, 1, 0) {
+		b.input = make(chan Transferable, b.batchSize)
+		go b.acceptInput()
+	}
```

Comment:
Better yet, I'm going to remove the manual `Reset()` call and just have `Add()` check for the condition. This will also avoid the situation where a panic would occur if an `Add()` is called after an `Exit()` but before a `Reset()`.